stan/math/rev/fun/multiply.hpp

```diff
@@ -554,7 +554,7 @@ class multiply_mat_vari<Ta, 1, Ca, double, 1> : public vari {
  * @return Product of scalar and matrix.
  */
 template <typename Mat1, typename Mat2,
-          require_all_eigen_t<Mat1, Mat2>* = nullptr,
+          require_all_eigen_vt<is_var_or_arithmetic, Mat1, Mat2>* = nullptr,
```

Comment:
Was it a problem if this was taking higher order autodiff types or something?

Reply:
This was just a cleanup because the multiply requires are very specialized and confusing